lecture/src/test/java/de/tum/in/www1/artemis/lecture/security/jwt/JWTFilterTest.java

```diff
@@ -0,0 +1,104 @@
+package de.tum.in.www1.artemis.lecture.security.jwt;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collections;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.mock.web.MockFilterChain;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import de.tum.in.www1.artemis.security.Role;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import tech.jhipster.config.JHipsterProperties;
+
+class JWTFilterTest {
+
+    private static final long ONE_MINUTE_MS = 60000;
+
+    private TestTokenGenerator testTokenGenerator;
+
+    private JWTFilter jwtFilter;
+
+    @BeforeEach
+    public void setup() {
+        JHipsterProperties jHipsterProperties = new JHipsterProperties();
```

Comment:
Do we need this whole Token thing everywhere? Would it be possible to import it somehow so we can define it at one place?

Reply:
I removed one line that could be removed

Reply:
While trying to refactor it I found out that the TokenProvider has changed in Artemis which made me try again to reuse the TokenProvider and the JwtFilter which if you remember didn't work with the gateway. I tried on both user-management and lecture and it worked. Therefore the tests are also removed and will not be needed in every microservice

With some changes in the configs I also managed to remove the token provider from the gateway. 

Now the token provider is in a single place(Artemis).